sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,779 @@
+# Preamble
+
+SIP Number: 012
+
+Title: Burn Height Selection for a Network Upgrade to Introduce New Cost-Limits
+
+Authors:
+* 0xAsteria <asteria@syvita.org>
+* Aaron Blankstein <aaron@hiro.so>
+* Diwaker Gupta <diwaker@hiro.so>
+* Hank Stoever <hank@oby.io>
+* Jason Lau <jason@okcoin.com>
+* Ludovic Galabru <ludo@hiro.so>
+* Trevor Owens <trevor@stacks.ac>
+* Xan Ditkoff <xan@daemontechnologies.co>
+
+Consideration: Governance, Technical
+
+Type: Consensus
+
+Status: Draft
+
+Created: 2021-10-08
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+The current Clarity cost limits were set very conservatively in Stacks 2.0:
+transactions with contract-calls frequently exceed these limits, which
+negatively affects transaction throughput. This SIP proposes an update to these
+cost-limits via a network upgrade and further, that the network upgrade be
+executed at a block height chosen by an off-chain process described in this SIP.
+
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0
+Universal license, available at
+https://creativecommons.org/publicdomain/zero/1.0/ This SIP’s copyright is held
+by the Stacks Open Internet Foundation.
+
+
+# Introduction
+
+The current Clarity cost limits were set very conservatively in Stacks 2.0.
+Since the mainnet launch on 2021-01-14, traffic on the network has grown
+steadily. And in recent months we have seen network congestion adversely
+impacting user experience on multiple occassions: valid transactions are not
+getting processed because they exceed one or more of the block limits, and are
+therefore not included by miners.
+
+Here are some relevant results from a [recent
+analysis](https://github.com/blockstack/stacks-blockchain/discussions/2883) of
+block data:
+
+* Average number of "events" per block: 34.9. This is lower than the demands of
```

Comment:
If you're going to report an average, then please also report a standard deviation.  Also, please explicitly state what an "event" is -- is this a Clarity call? A `(print )` statement? An asset transfer? Something else?

Reply:
Removed